### ISSUE TITLE
Duplicate classes generated after upgrading to Kotlin 1.6.21

### ISSUE THREAD

**Issue by efirestone** [square/wire#2327]
After updating from Kotlin 1.6.10 to 1.6.21 in my Android project the Kotlin classes get generated twice in two different directories.

With 1.6.10:
```
shared/build/generated/source/wire/<proto package path>
```

With 1.6.21:
```
shared/build/generated/source/wire/commonCommonMain/<proto package path>
shared/build/generated/source/wire/commonMain/<proto package path>
```
(note that "commonMain" wasn't included in the 1.6.10 generated path)

This is a Multiplatform project and the `wire {}` plugin usage is in the `shared` module's grade file:

```
wire {
    sourcePath {
        // From Maven Local
        srcJar("my.project:api:SNAPSHOT")
    }
    kotlin {
        includes = listOf(
            "my.project.api.client.*",
        )
    }
}
```

The wire plugin adds two Gradle tasks: `generateCommonMainProtos` and `generateProtos`. `generateCommonMainProtos` generates just the `commonMain/...` build directory, and `generateProtos` generates both directories.

This issue occurs with Wire version 4.0.1 and 4.4.1 (and likely in between). Reverting to Kotlin 1.6.10 fixes it with both Wire versions. The issue persists when using Kotlin 1.7.10 and 1.7.20-Beta.

**Comment by efirestone** [comment]
The logic looks to be [here](https://github.com/square/wire/blob/master/wire-library/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/kotlin/SourceRoots.kt#L107)

**Comment by efirestone** [comment]
I think this is probably an actual bug, but I'm able to work around it for now by using the `generateCommonMainProtos` task instead of the `generateProtos` task.

**Comment by kirillzh** [comment]
I'm seeing these tasks which look misconfigured
<img width="632" alt="Screen Shot 2022-10-05 at 3 56 43 PM" src="https://user-images.githubusercontent.com/5660531/194151111-e6ec487e-4c92-429a-bf6e-967c41240ae7.png">

**Comment by TrevorSStone** [comment]
My workaround currently is to add this to the bottom of my build.gradle.kts file for the project

```kotlin
tasks.whenTaskAdded {
    if (name == "generateCommonIosMainProtos" || name == "generateCommonCommonMainProtos") {
        enabled = false
    }
}
```

**Comment by oldergod** [comment]
Fixed with #2423
